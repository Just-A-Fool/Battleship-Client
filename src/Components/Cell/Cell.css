.cell {
  width: 90%;
  height: 10%;
  background-color:blue ;
  margin: 2px;
  font-size: 1rem;
  color: white;
  font-weight:bolder;
}

.label {
  background-color: rgb(43, 162, 236, 0.5);
}

.selected {
  background-color: red;
}

.hit {
  background-color: lightgreen
}

.miss {
  background-color: black;
}

.ship{
    background-color: grey;
}

.shot {
  background-color: white;
}

.ship-shot {
  background-color: red;
}

@media screen and (min-width: 1025px) {
  .cell {
    height: 12%;
    width: 97%;
    font-size: 1.2rem;
  }
}
