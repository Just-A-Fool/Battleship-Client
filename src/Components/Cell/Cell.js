import React from 'react';
import './Cell.css';

class Cell extends React.Component {

  static defaultProps ={
    selected:null,
    idNumber:-2,
    shipTiles:[],
  }

constructor(props){
  super(props);
    this.state = {
      x: this.props.x,
      y: this.props.y,
      occupied: null,
      hasBeenShot: null,
      index: this.findMyIndex, 
      isShipTile:false, 
  }
  //this.findMyIndex = this.findMyIndex.bind(this)
}

// componentDidMount(){
//   if ((this.props.shipTiles) && (this.props.shipTiles.indexOf(this.props.currentId)!==(-1))){
//     this.setState({shipTile:true})
//   }
//   this.checkForShipTile()
// }

// updateShipTile=(value, selected)=>{
//   console.log(value)
//   console.log(selected)
//   //console.log(this.state.x + this.state.y)
//   if(value === selected){
//     this.setState({shipTile : true})
//   }
// }

determineClassName = () => {
  //console.log(this.props.selected)
  //console.log(this.props.id)
  let className = 'cell';
  if(this.props.hits && this.props.hits.includes(this.props.id)){
    className += ' hit'
  } else
  if (this.props.misses && this.props.misses.includes(this.props.id)){
    className += ' miss'
  } else
  if(this.props.shipTiles.length > 0){
      if (this.props.shipTiles.indexOf(this.props.idNumber) !== (-1)){
         className = 'ship'
      }
  //this.setState({shipTile:true})
  //} 
    //this.setState({shipTile:true})
  } 
  // if ((this.props.selected === this.props.id) && (this.props.shipTiles) && 
  //    (this.props.shipTiles.indexOf(this.props.id)!==(-1) && 
  //    this.state.shipTile)){
  //       className = 'ship'
  //   //this.setState({shipTile:true})
  // } 
  else if (this.props.selected === this.props.id) {
    className += ' selected'
    //console.log(this.props.shipTiles)
  } 
  // if ((this.props.selected === this.props.id) && (this.props.shipTiles.length > 0) && (this.props.shipTiles.indexOf(this.props.id)!==(-1))){
  //     className += ' ship'
  // }
   //else if (this.props.selected && this.props.shipOccupied.indexOf(this.props.id)!==(-1)) {
    //className += ' ship'
  //}
  return className;
}

checkForShipTile = () => {
  console.log(this.props.shipTiles)
  console.log(this.props.currentId)
  if ((this.props.shipTiles) && (this.props.shipTiles.indexOf(this.props.currentId)!==(-1)) ){
   this.setState({isShipTile:true})
 }
 //this.render()
}

canCellBeClicked = () => {
  //console.log('here');
  if(this.props.hits && this.props.hits.includes(this.props.id)){
    return false;
  }
  else if (this.props.misses && this.props.misses.includes(this.props.id)){
    return false;
  }
  else if (this.props.label){
    return false;
  }
  else {
    return true;
  }
}

handleClick = () => {
  this.checkForShipTile()
  //console.log('yes');
  let click = this.canCellBeClicked();
  if(click){
    //this.checkForShipTile()
    this.props.handleSelectTarget(this.props.id, this.props.idNumber)
  }
  //this.checkForShipTile()
}

  render () {
    //console.log(this.props.id)
    return (
      //Adding in an id of label for the cells that are not part of the 
      //gameplay but are the labels for the board.
      <div id=
      {this.props.label 
        ? 'label' 
        : ''} 
        style={this.state.shipTile ? { 'backgroundColor': 'blue' } : null}
        className={this.determineClassName()}
        onClick={() => this.handleClick()}
                            >
        {this.props.id}
      </div>
    )
  }
}

export default Cell;
