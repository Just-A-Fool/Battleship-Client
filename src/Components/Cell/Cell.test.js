import React from 'react';
import ReactDOM from 'react-dom';
import Cell from './Cell';
import { shallow } from 'enzyme'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'

describe('Cell component', () => {

  it(`Renders without crashing`, () => {
    const div = document.createElement('div');
    ReactDOM.render(<Cell />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it.skip(`Displays a cell tile when rendered`, () => {
    const wrapper = shallow(<Cell />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })


  it(`Changes class to 'hit' if targeted and fired upon by opposing player while being a current ship tile`, () => {
    const hits = ['A1', 'B1']
    const id = 'A1'
    const wrapper = mount(<Cell hits={hits} id={id}/>)
    expect(wrapper.find('Cell').hasClass('hit'))
  })

  it(`Changes class to 'miss' if targeted and fired upon by opposing player while not a ship tile`, () => {
    const hits = ['A1', 'B1']
    const id = 'C1'
    const wrapper = mount(<Cell hits={hits} id={id}/>)
    expect(wrapper.find('Cell').hasClass('miss'))
  })

  it(`Changes class to 'selected' if selected by the player`, () => {
    const selected = 'A1'
    const id = 'A1'
    const wrapper = mount(<Cell selected={selected} id={id}/>)
    expect(wrapper.find('Cell').hasClass('selected'))
  })

  it(`Changes class to 'ship' if the id of the current cell is contained in the 'shipTileValues'
      array located in state of UserGrid Component`, () => {
        const shipTileValues = ['A1', 'A2', 'A3', 'A4']
        const id = 'A1'
        const wrapper = mount(<Cell shipTileValue={shipTileValues} id={id}/>)
        expect(wrapper.find('Cell').hasClass('ship'))
   })

   it(`Changes class to 'shot' if current cell id is located in the 'opponentShots' array located
      in the state of the UserGrid Component (inherently a ship cell)`, () => {
        const opponentShots = ['A1', 'A2', 'A3', 'A4']
        const id = 'A1'
        const wrapper = mount(<Cell opponentShots={opponentShots} id={id}/>)
        expect(wrapper.find('Cell').hasClass('shot'))
      })


  it.skip(`Cannot be clicked if currently a label` , () => {
    const label = true
    const id = 'A'
    const wrapper = mount(<Cell label={label} id={id}/>)
    wrapper.find('Cell').simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })

})