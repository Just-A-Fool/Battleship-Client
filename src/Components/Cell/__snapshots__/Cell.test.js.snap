// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell component Cannot be clicked if currently a label 1`] = `
<Cell
  id="A"
  idNumber={-2}
  label={true}
  selected={null}
  shipTileValues={Array []}
>
  <div
    className="cell"
    id="label"
    onClick={[Function]}
  >
    A
  </div>
</Cell>
`;

exports[`Cell component Displays a cell tile when rendered 1`] = `
<div
  className="cell"
  id=""
  onClick={[Function]}
/>
`;
